pidlock/__init__.py

```diff
@@ -45,35 +52,37 @@ def lock(self, name, wait=0, mininterval=1):
             name: (str) Name of the file to be used as lock
             wait: (float) How long (sec.) should it wait for lock to be released. Default is 0
             mininterval: (float) Minimum interval (sec.) to balance between performance and resource usage. Default is 1
+
         Raises:
             PIDLockedException: PID file is locked by another process
+
         Example:
             >>> with locker.lock(name='sleepy_script', wait=0, mininterval=1):
             >>>     time.sleep(10)
         """
-        if not path.isdir(self.lockdir): os.mkdir(self.lockdir)
-
-        pidfile = path.join(self.lockdir, name + ".pid")
+        lockdir = path.join(self.lockdir, name + "@" + getfqdn())
```

Comment:
I think you misunderstood me with regards the host.  The host name should be included in the `lock.pid` file not in the directory name.  The intent is to ensure the lock is honoured if the job is running on a different host, but not attempt to remove the lock if the current job is on a different host.

i.e.

* Job running on hostA
* Job starts on hostB
  *  detects running job, sees different host in `lock.pid`, doesn't attempt to verify if PID is still running, just waits
* Job on hostA completes
* Job resumes on hostB

vs

* Job 1 running on hostA, dies but doesn't cleanup `lock.pid` (and parent folder)
* Job 2 on hostA starts
* Sees _same_ hosts in `lock.pid`, confirms PID not found, cleans and attempts to create lock.